wordpress-mobile/WordPress-Android
# Stats revamp v2 latest post summary card

This PR updates Latest Post Summary card to new design as shown below
- Adds Latest Post with featured image (if set)
- Displays stats in three columns (and removes graph)

TODO:
- [x] [Corresponding FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2373) needs to be merged before merging this

Fixes #16424

| Current | New Design | Dark Theme |
|---|---|---|
|![Screenshot_20220429_151424](https://user-images.githubusercontent.com/990349/165888982-5503976d-5420-41f4-b077-88ca9cf78417.png)|![Screenshot_20220429_151749](https://user-images.githubusercontent.com/990349/165889003-ee4b2b7c-cbdc-419c-a8fd-4a17ed448648.png)|![Screenshot_20220429_151809](https://user-images.githubusercontent.com/990349/165889007-d569040c-58c2-4366-9f41-bfe3282a7926.png) |


To test:

Setup:

- Go to App Settings (Tap on Avatar at the top right hand corner on My Site -> Find App Settings)
- Select Debug Settings
- Find StatsRevampV2FeatureConfig under Features in development enable it and restart app

Test 1:

- Launch app
- Go to stats either using quick links or menu
- Ensure that it opens in Insights tab otherwise switch to Insights tab
- Verify that the new Latest Post Summary card is visible and matches the new designs above
- Ensure that featured image is displayed (if a feature image is set)
- View More on the top right corner is visible if there's any data on Views, Likes or Comments
- If data is not available it will display a Share action or Create Post if no latest post as before
- _If the Latest Post Summary card is not present.  Go to stats management (either through ⚙️ menu at the top right hand corner or Add new stats card at the bottom) and add the card and save then return to Insights._

Test 2:

- Turn on the dark theme on your device
- Check the card is displayed as in third image above

Test 3:

- Disable StatsRevampV2FeatureConfig and restart app
- Return to Stats and make sure Most Popular Time card is displayed as in first image above as before.


## Regression Notes
1. Potential unintended areas of impact
None.  Only existing Latest Post Summary card will be updated to new design with additions of a featured image

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
Updated unit tests

PR submission checklist:
